test/e2e/serial/tests/configuration.go

```diff
@@ -845,69 +849,152 @@ var _ = Describe("[serial][disruptive] numaresources configuration management",
 				}
 			}
 			Expect(targetedKC).ToNot(BeNil(), "there should be at least one kubeletconfig.machineconfiguration object")
+			numTargetedKCOwnerReferences := len(targetedKC.OwnerReferences)
 
 			kcObj, err := kubeletconfig.MCOKubeletConfToKubeletConf(targetedKC)
 			Expect(err).ToNot(HaveOccurred())
 			initialTopologyManagerPolicy := kcObj.TopologyManagerPolicy
 
-			if initialTopologyManagerPolicy != v1beta1.NoneTopologyManagerPolicy {
-				By("modifying topology manager policy under kubeletconfig to none")
-				mcpsInfo, err := buildMCPsInfo(fxt.Client, ctx, nroOperObj)
-				Expect(err).ToNot(HaveOccurred())
-				Expect(mcpsInfo).ToNot(BeEmpty())
+			var performanceProfile perfprof.PerformanceProfile
+			profiles := &perfprof.PerformanceProfileList{}
+			err = fxt.Client.List(ctx, profiles)
+			Expect(err).ToNot(HaveOccurred(), "unable to query performance profile")
 
-				Eventually(func(g Gomega) {
-					err := fxt.Client.Get(ctx, client.ObjectKeyFromObject(targetedKC), targetedKC)
+			for _, profile := range profiles.Items {
+				if profile.Kind == "PerformanceProfile" {
+					performanceProfile = profile
+				}
+			}
```

Comment:
I iterated over the profiles to get the first one. But I can see that I can even delete all that because everywhere in the code where I am changing the TM policy I'm querying it right there so this can actually be removed entirely.